dataset.go

```diff
@@ -40,11 +40,12 @@ var Record = &datasetType{}
 
 type datasetType struct{}
 
-func (sett *datasetType) String() string  { return sett.Name() }
-func (*datasetType) Name() string         { return "record" }
-func (*datasetType) Size() uint           { panic("call Size on each Data") }
-func (sett *datasetType) Data(n int) Data { return sett.DataEmpty(n) }
-func (*datasetType) DataEmpty(int) Data   { panic("use NewDataset function") }
+func (sett *datasetType) String() string      { return sett.Name() }
+func (*datasetType) Name() string             { return "record" }
+func (*datasetType) Size() uint               { panic("call Size on each Data") }
+func (sett *datasetType) Data(n int) Data     { return sett.DataEmpty(n) }
+func (*datasetType) DataEmpty(int) Data       { panic("use NewDataset function") }
+func (*datasetType) DataBuilder() DataBuilder { panic("use NewDatasetBuilder function") }
```

Comment:
why do we need to panic here? can't we use the following and remove NewDatasetBuilder?
```suggestion
func (*datasetType) DataBuilder() DataBuilder { return &datasetBuilder{} }
```

without it, you might fail when trying to append records

Reply:
I removed panic from `datasetType.Builder`, but left `NewDatasetBuilder` constructor as is since sometimes we need to create a builder before we have a concrete dataset.